hack/local-development/gcp/network.tf

```diff
@@ -0,0 +1,31 @@
+resource "google_compute_network" "dev-network" {
+  name                     = var.name
+  auto_create_subnetworks  = false
+  enable_ula_internal_ipv6 = false
+}
+
+resource "google_compute_subnetwork" "dev-subnetwork" {
+  name   = var.name
+  region = var.region
+
+  network          = google_compute_network.dev-network.id
+  stack_type       = "IPV4_IPV6"
+  ip_cidr_range    = "10.0.0.0/22"
+  ipv6_access_type = "EXTERNAL"
+}
+
+resource "google_compute_firewall" "dev-firewall" {
+  name    = var.name
+  network = google_compute_network.dev-network.name
+
+  source_ranges = ["0.0.0.0/0"]
```

Comment:
Generally, this would be a good idea. However, in our setup, we frequently switch networks in a single day (VPN on/off).
That's why we allow all incoming traffic on port 22.
Also, I don't think this is as critical as restricting access to a `Bastion` for example, as this is only a development machine and protected by ssh authentication.

Reply:
Development machines shouldn't be less protected as they can hold sensitive information like IaaS credentials needed for further testing.
I think it would be good to offer a secure by default like setup. You can still decide to overwrite the source range via Terraform var to match your case.

Reply:
Let's continue the discussion at https://github.com/gardener-community/dev-box-gcp/pull/1#discussion_r1083762756